### PR TITLE
Monitor awsarm through SSH

### PR DESCRIPTION
**Pull request by joinemm**
[sshified](https://github.com/hoffie/sshified) can be used to proxy prometheus metrics through ssh connection, meaning we do not have to open any ports to the world.

This PR adds a derivation to build `sshified`, a systemd service to run it, and the necessary ssh configuration to scrape metrics from awsarm.

Also added documentation that tells how to add new monitored targets, with either ssh or http authentication, though there are no more targets that are using http basic auth.

### CODE REVIEW

File: hosts/binarycache/configuration.nix [hunk 1]
```diff
@@ -56,7 +57,7 @@
     virtualHosts = {
       "cache.vedenemo.dev" = {
         enableACME = true;
-        forceSSL = true;
+        addSSL = true;
```
Comment (henrirosten) [inline hosts/binarycache/configuration.nix:60]: Why is this change needed again?

**Review by henrirosten — DISMISSED** [review]
Looks good to me.
One item we possibly might want to simplify in a a follow-up PR:
I think we do not necessarily need the `sshified` but similar ssh tunnelling setup could be established using just `ssh` with option `-L` (or possibly `-D`). Or did you trial this already?

**Comment by joinemm** [comment]
@henrirosten `ssh -L` works but the problem is we would have to keep the ssh tunnel open indefinitely, which is not very reliable. `sshified` on the other hand opens a new ssh connection for each request.

File: hosts/binarycache/configuration.nix (hunk 1 above)
Comment (joinemm) [reply hosts/binarycache/configuration.nix:60]: this was to allow http connections to cache (using ip address), but maybe shouldn't be part of this pr

Comment (joinemm) [reply hosts/binarycache/configuration.nix]: removed this for now

File: docs/monitoring-server.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+<!--
+SPDX-FileCopyrightText: 2023 Technology Innovation Institute (TII)
+
+SPDX-License-Identifier: Apache-2.0
```
Comment (tervis-unikie) [inline docs/monitoring-server.md:4]: Documentation license should be CC-BY-SA-4.0

**Review by tervis-unikie — COMMENTED** [review]
Please make sure that the sshfied user on awsarm only has the permissions absolutely necessary for this to work.

File: docs/monitoring-server.md (hunk 2 above)
Comment (joinemm) [reply docs/monitoring-server.md:4]: Good catch. Actually it looks like all of the other documentation in this repo is also using apache license which should be changed @henrirosten

**Comment by joinemm** [comment]
@tervis-unikie sshified user has whatever is default permissions when using `adduser`. Should something be restricted?

File: docs/monitoring-server.md (hunk 2 above)
Comment (henrirosten) [reply docs/monitoring-server.md:4]: True, we should change the documentation licenses. I propose we do this repo-wide with a separate PR after this has been merged.

**Review by henrirosten — APPROVED** [review]

**Comment by tervis-unikie** [comment]
> @tervis-unikie sshified user has whatever is default permissions when using `adduser`. Should something be restricted?

Then the user can do whatever a normal user can do, which for the purpose of sshfied is excessive. 
This account could be set into a limited chroot environment for example.
But anyway, this is not a matter of this PR, just noting this...

File: docs/monitoring-server.md (hunk 2 above)
Comment (tervis-unikie) [reply docs/monitoring-server.md:4]: I noticed this earlier this week, there's already Jira ticket about it.
